pkg/sql/internal.go

```diff
@@ -147,6 +147,7 @@ func newInternalExecutorWithTxn(
 		extraTxnState: &extraTxnState{
 			txn:                    txn,
 			descCollection:         descCol,
+			jobs:                   new(jobsCollection),
 			schemaChangeJobRecords: schemaChangeJobRecords,
```

Comment:
This is a confusing part that I introduced 😓 We should have had `schemaChangeJobRecords: make(map[descpb.ID]*jobs.Record)` here and remove it from the parameter list. We should also have added a comment that except descriptor collection, all metadata (such as jobs) is this internal executor's own set, thus their lifecycle is managed by the internal executor itself, and independent of the caller.

Reply:
The truth of the matter is that now that we're here, it seems like the `InternalExecutorFactory` should also be constructing the `descs.Collection` and should own its lifecycle. There's no reason for `descs.CollectionFactory.Txn` anymore. Honestly, that's sort of beautiful. We'll leave that for later. I moved the map construction for now.